src/main/kotlin/car/domain/Cars.kt

```diff
@@ -0,0 +1,31 @@
+package car.domain
+
+class Cars(private val cars: List<Car>, private val energyProvider: EnergyProvider = RandomEnergyProvider()) {
+
+    private val _histories: MutableList<CarMoveHistoryCollection> = ArrayList()
+    val histories: List<CarMoveHistoryCollection>
+        get() = _histories
+    val winners: Winners
+        get() = Winners(_histories.last())
+
+    constructor(
+        carNameCollection: CarNameCollection,
+        energyProvider: EnergyProvider = RandomEnergyProvider()
+    ) : this(carNameCollection.names.map { Car(it) }, energyProvider)
+
+    fun move(times: Int) {
+        repeat(times) {
+            moveAllCar()
+        }
+    }
+
+    private fun moveAllCar() {
+        val histories: List<CarMoveHistory> = cars.map {
+            it.move(energyProvider.getEnergy())
+
+            CarMoveHistory(it.name, it.currentPosition)
+        }.toList()
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
        }
```